metadata/containers.go

```diff
@@ -61,7 +61,64 @@ func (s *containerStore) Get(ctx context.Context, id string) (containers.Contain
 		return containers.Container{}, errors.Wrapf(err, "failed to read container %q", id)
 	}
 
-	return container, nil
+	if len(fieldmasks) == 0 {
+		return container, nil
+	}
+
+	ctrFieldMasked := new(containers.Container)
+	for _, path := range fieldmasks {
+		if strings.HasPrefix(path, "labels.") {
+			if ctrFieldMasked.Labels == nil {
+				ctrFieldMasked.Labels = map[string]string{}
+			}
+			key := strings.TrimPrefix(path, "labels.")
+			ctrFieldMasked.Labels[key] = container.Labels[key]
+			continue
+		}
+		if strings.HasPrefix(path, "specs.") {
+			if ctrFieldMasked.Spec == nil {
+				ctrFieldMasked.Spec = new(types.Any)
+			}
+			ctrFieldMasked.Spec = container.Spec
+			continue
+		}
+
+		if strings.HasPrefix(path, "extensions.") {
+			if ctrFieldMasked.Extensions == nil {
+				ctrFieldMasked.Extensions = map[string]types.Any{}
+			}
+			key := strings.TrimPrefix(path, "extensions.")
+			ctrFieldMasked.Extensions[key] = container.Extensions[key]
+			continue
+		}
+
+		switch path {
+		case "id":
```

Comment:
Breaking these out should be done by the protobuf definition, rather than manually.